src/components/ContactUs.js

```diff
@@ -1,72 +1,46 @@
+import { Modal, Button } from 'antd';
 import { navLink, TEXTS } from './constants';
+import FillForm from './SmallerComponents/Form';
 
-const { text1, text2, text3, text4, botton1, botton2 } = TEXTS;
-/*component for contact-us */
-class ContactUs extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            modal: false
-        };
-
-        this.toggle = this.toggle.bind(this);
-    }
+const { text1, submit, cancel, small, primary, back } = TEXTS;
+export default class ContactUs extends Component {
+  state = {
+    loading: false,
+    visible: false,
+  }
 
-    toggle() {
-        this.setState({
-            modal: !this.state.modal
-        });
-    }
+  showModal = () => {
+    this.setState({
+      visible: true,
+    });
+  }
 
-    render() {
-        return (
-            <div>
-                <div className={navLink} color="danger" onClick={this.toggle}>{text1}{this.props.buttonLabel}</div>
-                <Modal isOpen={this.state.modal} toggle={this.toggle} className={this.props.className}>
-                    <ModalHeader toggle={this.toggle}>{text1}</ModalHeader>
-                    <ModalBody>
-                        <Form>
-                            <FormGroup>
-                                <Label for="exampleName">{text2}</Label>
-                                <Input type="text" name="name" id="exampleName" placeholder="name here" />
-                            </FormGroup>
-                            <FormGroup>
-                                <Label for="exampleEmail">{text3}</Label>
-                                <Input type="email" name="email" id="exampleEmail" placeholder="email@example.com" />
-                            </FormGroup>
-                            <FormGroup>
-                                <Label for="exampleMessage">{text4}</Label>
-                                <Input type="textarea" name="message" id="exampleMessage" placeholder="text here" />
-                            </FormGroup>
-                        </Form>
-                    </ModalBody>
-                    <ModalFooter>
-                        <Button color="danger" onClick={this.toggle} email="exampleEmail.com">{botton1}</Button>{' '}
-                        <Button color="secondary" onClick={this.toggle}>{botton2}</Button>
-                    </ModalFooter>
-                </Modal>
-            </div>
-        );
-    }
-}
+  handleOk = () => {
+    this.setState({ loading: true });
+    setTimeout(() => {
+      this.setState({ loading: false, visible: false });
+    }, 3000);
+  }
 
-ContactUs.propTypes = {
-    buttonLabel: PropTypes.string,
-    className: PropTypes.string,
-};
+  handleCancel = () => {
+    this.setState({ visible: false });
+  }
 
-export default ContactUs;
+  render() {
+    const { visible, loading } = this.state;
+    return (
+      <div>
+        <div onClick={this.showModal} className={navLink} >{text1}</div>
```

Comment:
what does `text1` stand for? the variable name should be more descriptive. `text1` tells me nothing about the variable.

Reply:
Changed